src/Core/Auth/Utilities/DeviceExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Bit.Core.Entities;
+
+#nullable enable
+
+namespace Bit.Core.Auth.Utilities;
+
+public static class DeviceExtensions
+{
+    /// <summary>
+    /// Gets a boolean representing if the device has enough information on it to determine whether or not it is trusted.
```

Comment:
First off, love the method and the intent behind it!

However, for clarity, should this check be something like `hasTrustedDeviceKeys` vs `isTrusted`?  It's a bit misleading to call it is trusted when the trusted state of a device can only truly be determined client side with a valid device key to decrypt the `EncryptedPrivateKey` here and then have the privateKey decrypt the `encryptedUserKey`.  

If you would prefer to not change the name, it might be good to include that context in the summary at least explaining that the existence of the keys is the only thing that the server has to look at to determine if a device is likely to be trusted. 

In theory, these keys should never be populated unless the device was setup as a trusted device at least once but client side key persistence (e.g., user clears local storage) could make them irrelevant and make their device no longer trusted. 

Happy to discuss!

Reply:
Yes, it's not known to the server it's truest form of trust but I generally like the name, I can add a comment that explains that it's not fully known if it's trusted but to the server it's as known as it can be. 

Each time we use that method (I already use it on some other code) it's for guarding logic that we would only do for a "trusted" device so it feels like it reads better and would evolve better later on for if the logic of what server things a trusted device is. 

If you feel strongly I am more than willing to switch it up though.

Reply:
Done!